Thumb mode not detected on gcc for arm

efecanicoz-atlas opened the pull request:
#### 5528017994db81391afaba90dc2c1909d1d8842f
<pre>
Thumb mode not detected on gcc for arm
<a href="https://bugs.webkit.org/show_bug.cgi?id=300217">https://bugs.webkit.org/show_bug.cgi?id=300217</a>

Reviewed by Adrian Perez de Castro.

Add &quot;-mthumb&quot; flag to GCC to enable compiling thumb2 instruction set if supported.

* Source/cmake/OptionsCommon.cmake:

Canonical link: <a href="https://commits.webkit.org/301124@main">https://commits.webkit.org/301124@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2804f236536c7783b7cd66882fb5fe1c9b3a911

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125028 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44697 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35434 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131877 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76907 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45389 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53263 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95171 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127982 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36226 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111816 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75714 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/35144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75359 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117126 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105989 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30199 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134553 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123546 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51850 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39646 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103641 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52275 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108028 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103415 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26325 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48755 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27046 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/48897 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51735 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156571 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51109 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39212 "Found 8 new JSC stress test failures: wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.default-wasm, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-bbq, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-bbq-no-consts, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-collect-continuously, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-eager, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-eager-jettison, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-no-cjit, wasm.yaml/wasm/stress/wasm-js-string-builtins-returnStrictInt32.js.wasm-slow-memory (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54466 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52800 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d353defc70d692cb2392eb14545cf615047e0fe0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51928

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3d7a5ffabf6296247083b1a5e352f8a5ff2e175d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51928

aperezdc commented:
@efecanicoz-atlas One last change request: Could you please remove the tests line from the commit log? This one:

```
No new tests (OOPS!).
```

(While the merge-queue knows how to edit the “Reviewed by” line, it does not handle the tests one and will block the merge if it finds the “OOPS!” there.)

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e2804f236536c7783b7cd66882fb5fe1c9b3a911. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51928

efecanicoz-atlas commented:
> @efecanicoz-atlas One last change request: Could you please remove the tests line from the commit log? This one:
> 
> ```
> No new tests (OOPS!).
> ```
> 
> (While the merge-queue knows how to edit the “Reviewed by” line, it does not handle the tests one and will block the merge if it finds the “OOPS!” there.)

done

efecanicoz-atlas committed:
Thumb mode not detected on gcc for arm
https://bugs.webkit.org/show_bug.cgi?id=300217

Reviewed by Adrian Perez de Castro.

Add "-mthumb" flag to GCC to enable compiling thumb2 instruction set if supported.

* Source/cmake/OptionsCommon.cmake:

Canonical link: https://commits.webkit.org/301124@main

aperezdc commented:
@efecanicoz-atlas Thanks again for the fix, and for your patience with the contribution process 👏🏼

aperezdc commented:
Backported into `webkitglib/2.48` as commit 2e6d243429819f3e866a1c51b1afe7ab48d7e569

aperezdc commented:
Backported into `webkitglib/2.50` as commit 91a8191ab710aec66abeecb0c8263be6bf553d5b
